Rs aggregate count

jrchudy opened the pull request:
This PR adds functionality to the `recordset` directive to put a full entity count for the current reference at the top of the table. This can be seen in the `recordset` app and the `recordedit` foreign key popup selector.

This addresses issue #878.

jrchudy commented on common/table.js:
```diff
@@ -100,6 +100,7 @@
         */
         function read(scope, isBackground) {
 
+            if (scope.vm.search == '') scope.vm.search = null;
```
This is done so that the check on line 351 can compare like values. When the user uses back space to clear the search term, the search term field is `''` and the one on the new location object is set to `null` when it sees empty string as a search term.

RFSH commented on test/e2e/utils/recordedit-helpers.js:
```diff
@@ -613,6 +613,9 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
 
+                                        return chaisePage.recordsetPage.getTotalCount().getText();
+                                    }).then(function(text) {
+                                        expect(text).toBe("Displaying 5 of " + col.count + " Records", "The total count display in the foreign key popup is incorrect");
```
The way I wrote the test cases `col.count` was showing the number of rows that are visible. So it's not actually the total number. This shouldn't be a problem if the count is less than 25 (maximum page size allowed).  You can make a note of it in the jsdoc of this function, or add another `totalCount` variable.
Also change the exception message to use `colError`  which will add column name and index.

```javascript
// if you're adding the note that count should not be more than 25.
var expected = "Displaying " + col.count + " of " + col.count + " Records";

// if you're adding the new variable
var expected = "Displaying " + col.count + " of " + col.totalCount   + " Records";

expect(text).toBe(expected, colError(col.name, "The total count display in the foreign key popup is incorrect");
```
Is the maximum number of foreignkeys 25? if it's not please change it accordingly.

jrchudy commented:
@RFSH I addresses your comments and added another few test cases to make sure the displayed value is updated when the user triggers foreground searches.
